swift/ql/lib/codeql/swift/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -983,6 +983,13 @@ predicate readStep(Node node1, ContentSet c, Node node2) {
     node1.(DictionarySubscriptNode).getExpr() = subscript and
     c.isSingleton(any(Content::TupleContent tc | tc.getIndex() = 1))
   )
+  // read of an optional into the loop variable via foreach
```

Comment:
What is this for?  I gather some kind of automatic read of an optional (the result of `next()`?) inside the for-in loop CFG?

Reply:
Exactly - in the loop header the result of `next` is checked to see if it's `nil` or `some`, if it's `some` there's an implicit read before it's assigned to the loop variable.